portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,46 @@ function addRandomFact() {
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
 }
+
+/**
+ * Uses fetch() to obtain data from DataServlet.java
+ */
+ function obtain_fetch_data() {
+     fetch("/data")
+     .then(response => response.text())
+     .then(text => {
+         document.getElementById("fetch-div").innerText = text;
+     })
+     .catch(() => {
+         document.getElementById("fetch-div").innerText = "Error!";
+     })
+ }
+
+/**
+ * Uses fetch() to obtain json data from DataServlet.java
+ */
+ function obtain_json_data() {
+    fetch("/data")
+        .then(response => response.json())
+        .then(messages => {
+         console.log(messages)
```

Comment:
Format: add space.

Reply:
Also remove debugging logs.

Reply:
Can you remove console.log(messages)? It's used for debugging and shouldn't be submitted as part of the code.